sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/NestedColumnAliasingSuite.scala

```diff
@@ -738,6 +738,31 @@ class NestedColumnAliasingSuite extends SchemaPruningTest {
     val optimized = Optimize.execute(query)
     comparePlans(optimized, query)
   }
+
+  test("SPARK-35972: NestedColumnAliasing should consider semantic equality") {
+    val dataType = new StructType()
+      .add(StructField("itemid", StringType))
+      .add(StructField("search_params", StructType(Seq(
+        StructField("col1", StringType),
+        StructField("col2", StringType)
+      ))))
+    val relation = LocalRelation('struct_data.struct(dataType))
+    val plan = relation
+      .repartition(100)
+      .select(
+        GetStructField('struct_data, 1, None).as("value"),
```

Comment:
This is the key to trigger this bug. Ideally, in SQL query, nested columns should result to `GetStructField` with non-None `name`. But there are places that can create `GetStructField` with None `name`, such as `UnresolvedStar.expand`, Dataset encoder stuff, etc.

It's better to have an end-to-end test, but I'm fine with this UT.

Reply:
Yea, I see. For None case, the current `nestedFieldToAlias` cannot catch it up.